openmdao/components/tests/test_exec_comp.py

```diff
@@ -1337,8 +1337,10 @@ def test_register_check_partials_not_safe_mult_expr(self):
             p = om.Problem()
             comp = p.model.add_subsystem('comp', om.ExecComp(['out1 = unsafe(x) * z',
                                                               'out2 = safe(y) + z'], shape=size))
-            comp.declare_partials('out1', ['x', 'z'], method='fd')
-            comp.declare_partials('out2', ['y', 'z'], method='cs')
+            rows = cols = np.arange(size)
+            comp.declare_partials('out1', ['x', 'z'], rows=rows, cols=cols, method='fd')
+            comp.declare_partials('out2', ['y'], rows=rows, cols=cols, method='cs')
+            comp.declare_partials('out2', ['z'], rows=rows, cols=cols, method='fd')
```

Comment:
This was changed because we now enforce that we only call apply_nonlinear once for each column, so for a given 'wrt' variable, if we only compute the column once and the derivative of *any* 'of' variable with respect to our 'wrt' variable is not complex safe, then we have to use 'fd' for that column.